dashboard/app/controllers/api/v1/pd/workshop_survey_report_controller.rb

```diff
@@ -104,15 +111,32 @@ def local_workshop_survey_report
       end
     end
 
-    # GET /api/v1/pd/workshops/:id/local_workshop_daily_survey_report
-    def local_workshop_daily_survey_report
-      unless @workshop.local_summer? || @workshop.teachercon? ||
-        ([COURSE_CSP, COURSE_CSD].include?(@workshop.course) && @workshop.workshop_starting_date > Date.new(2018, 8, 1))
-        return render status: :bad_request, json: {
-          error: 'Only call this route for new academic year workshops, 5 day summer workshops, local or TeacherCon'
+    # GET /api/v1/pd/workshops/:id/generic_survey_report
+    def generic_survey_report
+      return local_workshop_daily_survey_report if @workshop.summer? ||
+        ([COURSE_CSP, COURSE_CSD].include?(@workshop.course) &&
```

Comment:
I'm not sure if we'll ever change which courses these are for, but we have CSD & CSP hardcoded in a lot of places in our source tree, when it feels like that list could be centralized as well.

Reply:
You mean centralizing all the validation logic using course names?

Reply:
I wonder if we can have a constant or a function that gets all relevant courses for this kind of situation?

One way to illustrate is what happens when we add an additional course.. we will have to find all these hardcoded references to CSD & CSP and ponder whether we add that additional course to all those places?  It seems it would be a better factored architecture to just add that new course to one central place.

Not a change for now, just a thought.